openlabclient/openlabclient/cli.py

```diff
@@ -0,0 +1,109 @@
+import argparse
+import sys
+import yaml
+
+from openlabclient.plugins import base
+from openlabclient import utils
+from openlabclient.utils import _color
+
+
+class OpenLabCmd(object):
+    def __init__(self):
+        self.parser = None
+        self.args = None
+
+    def create_parser(self):
+        parser = argparse.ArgumentParser(
+            description='The command line tool for OpenLab management',
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+        parser.add_argument('-c', dest='config',
+                            default='/etc/openlab/openlab.yaml',
+                            help='path to config file')
+
+        subparsers = parser.add_subparsers(title='commands',
+                                           description='valid commands',
+                                           dest='command',
+                                           help='additional help')
+
+        cmd_check = subparsers.add_parser('check',
+                                          help='check openlab environment')
+        cmd_check.set_defaults(func=self.check)
+        cmd_check.add_argument('--type', default='all',
+                               help="Specify a plugin type, like 'nodepool', "
+                                    "'jobs'. Default is 'all'.")
+        cmd_check.add_argument('--cloud', default='all',
+                               help="Specify a cloud provider, like 'otc', "
+                                    "'vexxhost'. Default is 'all'.")
+        cmd_check.add_argument('--nocolor', action='store_true',
+                               help='Enable the no color mode.')
+        cmd_check.add_argument('--recover', action='store_true',
+                               help='Enable the auto recover mode.')
+
+        return parser
+
+    def _get_cloud_list(self, cloud):
+        with open('/etc/openstack/clouds.yaml') as f:
```

Comment:
I see a `-c` option in line#20, this is hard code?

Reply:
This cloud config is different from the openlab config, In future, we can use:
```
openlab check --config /etc/openlab/openlab.yaml
```
and this ``'/etc/openstack/clouds.yaml'`` can be set in the conif like
```
cat  /etc/openlab/openlab.yaml
[default]
cloud_yaml = /etc/openstack/clouds.yaml
```

Reply:
++ for @Yikun .While this pr is only for interface refactor.  `-c` doesn't work at this moment. This is just a placeholder. We can cooperate to update the tools after this refactor.

Reply:
It's OK to me.